src/analysis/convex/basic.lean

```diff
@@ -423,6 +423,21 @@ begin
   exact h hx hy ha hb hab
 end
 
+lemma convex_iff_forall_pos_ne :
+  convex 𝕜 s ↔ ∀ ⦃x y⦄, x ∈ s → y ∈ s → x ≠ y → ∀ ⦃a b : 𝕜⦄, 0 < a → 0 < b → a + b = 1
+  → a • x + b • y ∈ s :=
+begin
+  refine ⟨λ h x y hx hy _ a b ha hb hab, h hx hy ha.le hb.le hab, _⟩,
+  intros h x y hx hy a b ha hb hab,
+  obtain rfl | ha' := ha.eq_or_lt,
+  { rw [zero_add] at hab, simp [hab, hy] },
+  obtain rfl | hb' := hb.eq_or_lt,
+  { rw [add_zero] at hab, simp [hab, hx] },
+  obtain rfl | hxy := eq_or_ne x y,
+  { rwa convex.combo_self hab },
+  exact h hx hy hxy ha' hb' hab,
+end
```

Comment:
How about this version?
```suggestion
lemma convex_iff_pairwise_on :
  convex 𝕜 s ↔ s.pairwise_on (λ x y, ∀ ⦃a b : 𝕜⦄, 0 < a → 0 < b → a + b = 1 → a • x + b • y ∈ s) :=
begin
  refine ⟨λ h x hx y hy _ a b ha hb hab, h hx hy ha.le hb.le hab, _⟩,
  intros h x y hx hy a b ha hb hab,
  obtain rfl | ha' := ha.eq_or_lt,
  { rw [zero_add] at hab, simp [hab, hy] },
  obtain rfl | hb' := hb.eq_or_lt,
  { rw [add_zero] at hab, simp [hab, hx] },
  obtain rfl | hxy := eq_or_ne x y,
  { rwa convex.combo_self hab },
  exact h _ hx _ hy hxy ha' hb' hab,
end
```
This gives you access to some API for `pairwise`, which might be handy; and is defeq up to permutation of binders.